lib/user.rb

```diff
@@ -0,0 +1,27 @@
+class User
+
+  attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(jokester, joke)
+    jokester.learn(joke)
+  end
+
+  def joke_by_id(id)
+    @jokes.find do |joke|
+      joke.id == id
```

Comment:
perfect use case for this specific enumerable! great work choosing the right tool for this job.